### ISSUE TITLE
[Feature Request] Warn about multiple assignments in one line

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1110]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Feature Request                          |
| Plugin         | Php Inspections (EA Ultimate), 2.0.9 |
| Language level | PHP 7.2                        |

Current behaviour:
---

Here is some example code:

```php
<?php
$foo = $bar = false;
```

No warning given.

Suggested behaviour:
---

Add an inspection that warns about multiple assignments in the same line. For me personally it doesn't look like a best practise at all and is just confusing.

**Comment by machour** [comment]
I did some digging and the only references I could find about this being best practice or not is this thread: https://groups.google.com/forum/#!topic/php-fig/JS5Z8fVqLSI

But this only concerns properties declaration in a class.

I personally like the ability to declare multiple variables on a single line, when the value is of a simple type (number, string, boolean, empty array ..) but tend to avoid it when dealing with objects.

**Comment by funivan** [comment]
My rule: avoid it in all cases.

**Comment by andreasschroth** [comment]
@machour Thanks for the link, interesting to read, although no real outcome over there as well.

Personally, I'd love to see it as an optional inspection that's deactivated by default - form what I can see some people like multiple assignments per line, while others avoid them at all costs. I don't want to force any developer or team to avoid them, if they like them and it works well for them. But on the other side it would be good if we could turn the inspection on for teams who don't want to use multiple assignments in the same line.

**Comment by kalessil** [comment]
Implemented
